### ISSUE TITLE
About LPF-SGD implementation

### ISSUE THREAD

**Issue by lucasliunju** [devansh20la/LPF-SGD#1]
Thanks so much for releasing the code. I have several questions about the implementation about LPF-SGD.

1. `noise.append(- init_mp - temp)` the reason that using init_mp to obtain the noise

2.  `mp.grad.add_((-(n**2 + 1) / mp.view(-1).norm().item())*batch_loss.item())` why we still need this value to the gradient.

**Comment by devansh20la** [comment]
I think you might be looking at the wrong file. This was from another optimizer.

**Comment by lucasliunju** [comment]
Thank you very much for your reply!

**Comment by lucasliunju** [comment]
Dear Devansh,

This is a great work for me and LPF-SGD can significantly improve the performance compared with vanilla momentum sgd. I would like to ask whether cosine learning rate decay can also work for LPF-SGD on WRN and ResNet. Since I find the most learning rate decay is `StepLR`

Thank you very much!

Best,
Lucas

**Comment by devansh20la** [comment]
I did not try the cosine learning rate decay. I can't think of any reason it shouldn't work, it might just require finetuning to find the best hyperparameters.

**Comment by lucasliunju** [comment]
Dear Devansh,
Thank you very much for your reply. I have reproduced the results based on cosine decay, which is also the same as the results in the paper. Thanks again!

Best,
Lucas
